docs/ocs-on-a-disconnected-environment.md

```diff
@@ -53,129 +53,148 @@ you should get something similar to this:
+
+
+- **Add the images we mirrored in `mapping.txt` to the imageContentSourcePolicy**  
+After `oc adm catalog mirror` is completed it will print an output dir where an `imageContentSourcePolicy.yaml` is generated (usually the catalog image with `-manifests` suffix. e.g: `redhat-operators-manifests`).
+the missing related images should be added to the `ImageContentSourcePolicy` spec. e.g. an entry for rhceph-rhel8: 
+  ```yaml
+    - mirrors:
+      - <your_registry>/ocs4/rhceph-rhel8
+      source: registry.redhat.io/ocs4/rhceph-rhel8
+  ```
+
+
+- **After adding the related images (required for OCS-4.3 only), apply the `imageContentSourcePolicy.yaml` file to the cluster**  
+(the file is generated by `oc adm catalog mirror`. the output dir is usually `./[catalog image name]-manifests`)
+  ```
+  oc apply -f ./[output dir]/imageContentSourcePolicy.yaml
+  ```
+
+
+### lib-bucket-provisioner workarounds
+
+- OCS versions 4.3 and 4.4 are still dependent on lib-bucket-provisioner which is part of community-operators catalog. To work around this dependency it's necessary to build a catalog image for the comminity-operators. The dependency is only on OB\OBC CRDs which are installed with the CSV (which is included in the catalog image), so there is no need to mirror the catalog.
+
+  - **build oeprators catalog for community operators**  
+This step is required as workaround for lib-bucket-provisioner dependency.  
+(again, the tag of the `origin-operator-registry` in the `--from` should match the major and minor versions of the OCP cluster. e.g. 4.3)
+
+  ```
+  oc adm catalog build --insecure --appregistry-endpoint https://quay.io/cnr --appregistry-org community-operators --from=quay.io/openshift/origin-operator-registry:4.3 --to=${MIRROR_REGISTRY_DNS}/olm/community-operators:v1 --registry-config=${AUTH_FILE}
+  ```
+
+  - **Create CatalogSource for community-operators**  
+Create a CatalogSource object that references the catalog image for community-operators. Modify the following to your specifications and save it as a catalogsource.yaml file:
+  ```yaml
+  apiVersion: operators.coreos.com/v1alpha1
+  kind: CatalogSource
+  metadata:
+    name: community-operators-catalog
+    namespace: openshift-marketplace
+  spec:
+    sourceType: grpc
+    image: <your_registry>/olm/community-operators:v1 
+    displayName: Community Operators Catalog
+    publisher: grpc
+  ```
+  create the catalog source:
+  ```
+  oc create -f catalogsource.yaml
+  ```
+
+- Prior to OCS 4.5 for OCP disconnected environments, the lib-bucket-provisioner csv and deployment will need to be edited and the image quay.io/noobaa/pause will need to be replaced with quay.io/noobaa/pause@sha256:b31bfb4d0213f254d361e0079deaaebefa4f82ba7aa76ef82e90b4935ad5b105. 
+After the installation of OCS, edit the lib-bucket-provisioner csv first with this image@sha. Next, edit the lib-bucket-provisioner deployment and replace quay.io/noobaa/pause with this image@sha if not already correct.
+
+
```

Comment:
From previous attempts. Creating a community operator catalogue source does not work. The catalogue pod spins up and complains about the db being in a read only mode. The pod stays healthy but the catalogue does not show up in OLM list. I have confirmation from two other consultants that this is the case and it is not one off. The only workaround that has worked for us is the customer catalogue created for libbucket (http://quay.io/noobaa/lib-bucket-catalog@sha256:b9c9431735cf34017b4ecb2b334c3956b2a2322ce31ac88b29b1e4faf6c7fe7d)

Reply:
I was facing the same issue with redhat-operators catalog previously and it takes me very long time to find out how to fix it. Still I'm not 100% sure, what exactly was the problem, but most probably it was related to the image used in command `oc adm catalog build ...` (`--from` argument) with combination with OCP server/client version. In the previous version of this document, this argument was missing in the command and I've also tried few variants found in some other docs and it behaves the same way. Not sure, where I've found the currently used image `quay.io/openshift/origin-operator-registry:4.4`, which worked for me with OCP 4.4.4.

Reply:
Thanks, @arvin-a. I'll change